pennylane/ops/qubit/special_unitary.py

```diff
@@ -0,0 +1,347 @@
+    The identity Pauli word :math:`I^{\otimes n}` does not belong to :math:`\mathfrak{su}(N)`
+    and therefore is not included.
+
+    The basis elements are ordered (choose the description that suits you most)
+
+      - lexicographically.
+
+      - such that the term acting on the last qubit changes fastest, the one acting on the first
+        qubit changes slowest when iterating through the output.
+
+      - such that the basis index, written in base :math:`4`, contains the indices for the list
+        ``["I", "X", "Y", "Z"]``, in the order of the qubits
+
+      - such that for three qubits, the first basis elements are the Pauli words
+        ``"IIX", ""IIY", "IIZ", "IXI", "IXX", "IXY", "IXZ", "IYI"...``
+
+    .. note::
+
+        Note that this method internally handles a complex-valued tensor of size
+        ``(4**num_wires, 2**num_wires, 2**num_wires)``, which requires at least
+        ``4 ** (2 * num_wires - 13)`` GB of memory (at default precision).
+
+    **Example**
+
+    >>> pauli_basis(1)
+    array([[[ 0.+0.j,  1.+0.j],
+            [ 1.+0.j,  0.+0.j]],
+           [[ 0.+0.j, -0.-1.j],
+            [ 0.+1.j,  0.+0.j]],
+           [[ 1.+0.j,  0.+0.j],
+            [ 0.+0.j, -1.+0.j]]])
+
+    >>> pauli_basis(3).shape
+    (63, 8, 8)
+    """
+    if num_wires < 1:
+        raise ValueError(f"Require at least one wire, got {num_wires}.")
+    if num_wires > 7:
+        raise ValueError(
+            f"Creating the Pauli basis tensor for more than 7 wires (got {num_wires}) is deactivated."
+        )
+    return reduce(np.kron, (_pauli_matrices for _ in range(num_wires)))[1:]
+
+
+@lru_cache
+def pauli_words(num_wires):
+    r"""Compute all :math:`n`-qubit Pauli words except ``"I"*num_wires``,
+    corresponding to the Pauli basis of the Lie algebra :math:`\mathfrak{su}(N)`.
+
+    Args:
+        num_wires (int): The number of wires, or number of letters per word.
+
+    Returns:
+        list[str]: All Pauli words on ``num_wires`` qubits, except from the identity.
+
+    There are :math:`d=4^n-1` Pauli words that are not the identity. They are ordered
+    (choose the description that suits you most)
+
+      - lexicographically.
+
+      - such that the term acting on the last qubit changes fastest, the one acting on the first
+        qubit changes slowest when iterating through the output.
+
+      - such that the basis index, written in base :math:`4`, contains the indices for the list
+        ``["I", "X", "Y", "Z"]``, in the order of the qubits
+
+      - such that for three qubits, the first Pauli words are
+        ``"IIX", ""IIY", "IIZ", "IXI", "IXX", "IXY", "IXZ", "IYI"...``
+
+    **Example**
+
+    >>> pauli_words(1)
+    ['X', 'Y', 'Z']
+    >>> len(pauli_words(3))
+    63
+    """
+    return ["".join(letters) for letters in product(_pauli_letters, repeat=num_wires)][1:]
+
+
+def special_unitary_matrix(theta, num_wires):
```

Comment:
Could we move this functionality inside the `SpecialUnitary.matrix` method?

Reply:
This is a fair question. I guess we can move it inside, yes.
I don't think we "save" passing around parameters as long as we're using the static methods, but it still might be nicer to make everything part of the class. For now I do not see any other purpose for `special_unitary_matrix`, and even less so for the `get_one_parameter_...` methods added in the later PR.

Reply:
Done.